src/scripts/themed_posts.js

```diff
@@ -0,0 +1,67 @@
+import { buildStyle, filterPostElements, blogViewSelector } from '../util/interface.js';
+import { getPreferences } from '../util/preferences.js';
+import { onNewPosts } from '../util/mutations.js';
+import { timelineObject } from '../util/react_props.js';
+
+const styleElement = buildStyle();
+const blogs = new Set();
+let blacklist;
+
+const defaultBackgroundColor = '#FAFAFA';
+
+const hexToRGB = (hex) => {
+  const chars = hex.replace('#', '').split('');
+  const splitnum = chars.length > 3 ? 2 : 1;
+  const arr = [];
+  while (chars.length) {
+    arr.push(parseInt(chars.splice(0, splitnum).join(''), 16));
+  }
+  return arr.join(', ');
+};
```

Comment:
i see the spirit of what this is trying to do, but i think a regex approach could be a lot more fun...

patch notes:
- made the function shorter in terms of sloc
- probably made the function longer in terms of total characters
- now supports `#RGB` syntax properly
- now supports alpha values even less! (`#RGBA` will be interpreted as `#RRGB`, `#RRGGBBAA` will throw an error)

```suggestion
const hexToRGB = (hex) => {
  const { red, green, blue } = hex.match(/^#(?<red>[A-Fa-f0-9]{1,2})(?<green>[A-Fa-f0-9]{1,2})(?<blue>[A-Fa-f0-9]{1,2})$/).groups;
  return [red, green, blue]
    .map(color => color.padEnd(2, color))
    .map(color => parseInt(color, 16))
    .join(', ');
};
```

we could support alpha values properly by having four perfect regexes (one per valid hex syntax) but i think it's safe enough to operate on the assumption that blog theme colours will never support alpha

Reply:
Wait, how does this - oh, greedy quantifiers give back. Wild.

(Also, whoops, I didn't realize I wasn't correctly handling `#RGB` colors. That is a weird but convenient shorthand!)

Adding a `{0,2}` alpha term would work for `#RRGGBBAA` but still parse `#RGBA` as `#RRGB`, right? Darn.

Reply:
I guess one could do `hex.replace(`#`, ``).slice(0, 6).match(` to at least not crash on... ah, still doesn't work for #RGBA, never mind.